app/views/traces/_trace.html.erb

```diff
@@ -1,35 +1,44 @@
+    <% if trace.inserted %>
+      <a href="<%= url_for :controller => "traces", :action => "show", :id => trace.id, :display_name => trace.user.display_name %>">
+        <img src="<%= url_for :controller => "traces", :action => "icon", :id => trace.id, :display_name => trace.user.display_name %>" alt="" class="img-thumbnail mr-3" />
+      </a>
+    <% else %>
+      <div style="width: 60px; height: 60px" class="img-thumbnail bg-white mr-3"></div>
+    <% end %>
+  <% end %>
+
+  <div class="media-body">
+    <% if trace.inserted %>
+      <p class="float-right">
+        <%= link_to t(".view_map"), { :controller => "site", :action => "index", :mlat => trace.latitude, :mlon => trace.longitude, :anchor => "map=14/#{trace.latitude}/#{trace.longitude}" }, { :class => "mr-2 border-right pr-2" } %>
+        <%= link_to t(".edit_map"), { :controller => "site", :action => "edit", :gpx => trace.id } %>
+      </p>
     <% end %>
-  </td>
-  <td><%= link_to trace.name, :controller => "traces", :action => "show", :display_name => trace.user.display_name, :id => trace.id %>
-    <span class="text-muted" title="<%= trace.timestamp %>"> ...
+
+    <p class="mb-0">
+      <strong class="mr-2">
+        <%= link_to trace.name, :controller => "traces", :action => "show", :display_name => trace.user.display_name, :id => trace.id %>
+      </strong>
       <% if trace.inserted %>
-        (<%= t ".count_points", :count => trace.size %>)
+        <%= t ".count_points", :count => trace.size %>
+      <% else %>
+        <span class="text-warning"><%= t ".pending" %></span>
       <% end %>
-      ... <%= time_ago_in_words(trace.timestamp, :scope => :'datetime.distance_in_words_ago') %></span>
-      <%= link_to_if trace.inserted?, t(".map"), { :controller => "site", :action => "index", :mlat => trace.latitude, :mlon => trace.longitude, :anchor => "map=14/#{trace.latitude}/#{trace.longitude}" }, { :title => t(".view_map") } %> /
-      <%= link_to t(".edit"), { :controller => "site", :action => "edit", :gpx => trace.id }, { :title => t(".edit_map") } %>
+      <% badge_class = trace.visibility.in?("public", "identifiable") ? "success" : "danger" %>
+      <span class="ml-2 badge badge-<%= badge_class %> text-white"><%= t(".#{trace.visibility}") %></span>
+    </p>
 
-      <% badge_class = case trace.visibility
-                       when "public", "identifiable" then "success"
-                       else "danger"
-                       end %>
-      <span class="badge badge-<%= badge_class %> text-white"><%= t(".#{trace.visibility}") %></span>
+    <p class="mb-0 text-muted" title="<%= trace.timestamp %>">
+      <%= time_ago_in_words(trace.timestamp, :scope => :'datetime.distance_in_words_ago') %>
+      <%= t ".by" %> <%= link_to trace.user.display_name, user_path(trace.user) %>
+      <% if trace.tags.present? %>
+        <%= t ".in" %>
+        <%= safe_join(trace.tags.collect { |tag| link_to_tag tag.tag }, ", ") %>
+      <% end %>
```

Comment:
This bit here looks like a Lego-style translation, as it translates the text word by word. Obviously I can't tell if that's causing some issue for any of the ~97 languages we support.

Typically you would define one locale string that includes multiple variables and let the translators figure out what the proper text would be in their language.

Reply:
I try not to touch the output – translations and things like this join – in all my PRs since my focus is on refactoring the html/css to bootstrap and while doing this moving the elements to a – in my opinion – better place in the view. I hope this will make it easier to merge since I expect less discussion about the right wording and less re-translating needs to be done.

If we where to change this specific translation, I would vote for changing the meaning of "in" to something like "categoriesed with" or "tagged with". The "in" term does not make much sense IMO, since the tags are not the main "folder" where the traces are "in", nor are the tags the main navigation element on the traces page.

Reply:
Yes that's a separate issue and should be addressed (if there is indeed a problem) in a separate PR.